source/common/ssl/connection_impl.cc

```diff
@@ -214,6 +214,22 @@ Network::ConnectionImpl::IoResult ConnectionImpl::doWriteToSocket() {
 
 void ConnectionImpl::onConnected() { ASSERT(!handshake_complete_); }
 
+bool ConnectionImpl::isMtls() {
+  if (SSL_get_peer_certificate(ssl_.get())) {
```

Comment:
Is it an invariant that if we've received a client cert that this cert has been appropriately verified and trusted?